app/src/main/java/org/freedombox/freedombox/views/fragments/DiscoveryFragment.kt

```diff
@@ -18,18 +18,37 @@
 package org.freedombox.freedombox.views.fragments
 
 import android.content.Context.NSD_SERVICE
+import android.content.Intent
 import android.content.SharedPreferences
 import android.net.nsd.NsdManager
 import android.net.nsd.NsdServiceInfo
 import android.os.Bundle
+import android.preference.PreferenceManager
 import android.util.Log
+import android.view.View
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import kotlinx.android.synthetic.main.fragment_discovery.*
 import org.freedombox.freedombox.R
 import org.freedombox.freedombox.components.AppComponent
+import org.freedombox.freedombox.views.activities.LauncherActivity
+import org.freedombox.freedombox.views.adapter.DiscoveryListAdapter
+import org.freedombox.freedombox.views.model.ConfigModel
 import javax.inject.Inject
 
 class DiscoveryFragment : BaseFragment() {
     val TAG = "DISCOVERY_FRAGMENT"
 
+    lateinit var adapter: DiscoveryListAdapter
+
+    var discoveredBoxList: ArrayList<String> = ArrayList<String>()
```

Comment:
Done

Reply:
@lispyclouds For class variables that are initialized later, I'd recommend using vars with immutable collections, instead of using vals with mutable collections.
Kotlin method parameters (just like Scala from which the whole lang is copied) are vals by default. So, the risk is in the collections getting mutated by the functions that are passed to them, not reassignment.

@abilaashsai Please use var + immutable collections wherever initialization is done later.